pkg/api/util/util.go

```diff
@@ -11,19 +11,24 @@ import (
 	"net"
 	"net/http"
 	"strings"
+	"sync"
 
 	"github.com/DataDog/datadog-agent/pkg/api/security"
 	"github.com/DataDog/datadog-agent/pkg/config/model"
 )
 
 var (
-	token    string
-	dcaToken string
+	tokenLock sync.RWMutex
+	token     string
+	dcaToken  string
 )
 
 // SetAuthToken sets the session token
 // Requires that the config has been set up before calling
 func SetAuthToken(config model.Reader) error {
+	tokenLock.Lock()
+	defer tokenLock.Unlock()
```

Comment:
Could you remove the comment a few lines down that says "token is only set once, no need to mutex protect".

Reply:
Sure haha, I didn't notice that